### DISCUSSION TITLE
GitHub Desktop asking for the credentials frequently.

### DISCUSSION THREAD

**Discussion by sunilvjr** [community/community#126208] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I am using GitHub Desktop to access code from the repository with user credentials but in the past 10-15 days, it is starting to ask for the credentials frequently. Like if I tried to pull the code after 10-15 min of successful login,I was asked for the credential again. Do other user facing the same issue?

**Comment by AsmitaMishra24** [comment]
Hey, @sunilvjr,

If GitHub Desktop is repeatedly asking for your credentials, especially after a short period of successful login, you’re not alone. Many users encounter this issue. Let’s address it:

Cause:
- The problem occurs when you use the HTTPS clone URL for your repository.
- HTTPS URLs prompt you to enter your GitHub username and password every time you pull or push a repository.

Solution:
- You can configure Git to store your password securely so that it doesn’t ask for credentials repeatedly.
- Here are two approaches: 

a. Update the URL to Use SSH:

- Change the remote URL of your repository to use SSH instead of HTTPS.
- Open your Git Bash or command line.
- Run the following command (replace username/repo with your actual repository):

`git remote set-url origin git@github.com:username/repo.git`

- This way, you won’t be prompted for credentials when interacting with the repository.

b. Store Credentials:

- Configure Git to store your username and password:

`git config --global credential.helper store`

- This will save your credentials, and Git won’t ask for them again.
- Alternatively, you can set a timeout for the stored credentials:

`git config --global credential.helper 'cache --timeout=600'`

- This caches your credentials for 10 minutes (adjust the timeout as needed).

Security Consideration:
- While storing credentials can be convenient, consider using SSH for better security.
- SSH allows you to connect to GitHub without supplying your username or password every time.

Hope this will be helpful!
